cli/test/e2e/package_authoring_e2e_test.go

```diff
@@ -293,24 +296,50 @@ func newPromptOutput(t *testing.T) promptOutput {
 	outputReader, outputWriter, err := os.Pipe()
 	require.NoError(t, err)
 
-	return promptOutput{t, stringWriter, stringReader, outputWriter, outputReader}
+	bufferedStdout := bytes.Buffer{}
+	bufferedWriter := io.MultiWriter(outputWriter, &bufferedStdout)
+
+	return promptOutput{t, stringWriter, stringReader, outputWriter,
+		outputReader, bufferedWriter, &bufferedStdout}
 }
 
 func (p promptOutput) WritePkgRefName() {
 	p.stringWriter.Write([]byte("afc.def.ghi\n"))
 }
 
-func (p promptOutput) OutputWriter() io.Writer { return p.outputWriter }
-func (p promptOutput) OutputReader() io.Reader { return p.outputReader }
-func (p promptOutput) StringWriter() io.Writer { return p.stringWriter }
-func (p promptOutput) StringReader() io.Reader { return p.stringReader }
+func (p promptOutput) OutputWriter() io.Writer   { return p.outputWriter }
+func (p promptOutput) OutputReader() io.Reader   { return p.outputReader }
+func (p promptOutput) StringWriter() io.Writer   { return p.stringWriter }
+func (p promptOutput) StringReader() io.Reader   { return p.stringReader }
+func (p promptOutput) BufferedWriter() io.Writer { return p.bufferedWriter }
 
 func (p promptOutput) Write(val string) {
 	p.stringWriter.Write([]byte(val + "\n"))
 }
 
 func (p promptOutput) WaitFor(text string) {
-	// Adding hardcode wait as waiting on text turns out to be flaky.
-	// TODO need to revisit this logic
-	time.Sleep(10*time.Second)
+	attempts := 0
+	// Poll buffered output till desired string is found
+	for {
+		found := strings.Contains(p.bufferedStdout.String(), text)
+
+		if os.Getenv("KCTRL_DEBUG_BUFERED_OUTPUT_TESTS") == "true" {
+			fmt.Printf("\n==> \t Buffered Output (Waiting for '%s' | found=%t)\n", text, found)
+			fmt.Println(p.bufferedStdout)
+			fmt.Printf("=\t=\t=\t=\t=\t=\t=\t=\t=\t=\n\n")
```

Comment:
I think we can improve it although I understand it will be disabled by default.

Reply:
Yeah I put this in here just to help debug flaky tests, especially since we plan on adding more